src/VASSAL/counters/BasicPiece.java

```diff
@@ -604,6 +652,21 @@ else if (oldMap != null) {
       }
     }
     setGpId(st.nextToken(""));
+    
+    if (persistentProps == null) {
+      persistentProps = new HashMap<>();      
+    }
+    else {
+      persistentProps.clear();
+    }
+      
+    // Persistent Property values will always be String (for now).
```

Comment:
could make persistentProps a Map<String, String>

Reply:
Yes, I thought about that, however the setProperty/getProperty interface is Object, Object and there is no reason why me have a need for other Objects in the future. Better to keep it generalised.

Reply:
no reason why we may NOT have a need for other objects :)